### PR TITLE
[HttpKernel] Remove old container files

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23872
| License       | MIT
| Doc PR        | -

ping @mvrhov

### CODE REVIEW

**Review by chalasr — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Status: needs work

If the new container is the same as the old, we shouldn't remove the files.

**Comment by nicolas-grekas** [comment]
Fixed
Status: needs review

**Review by ogizanagi — APPROVED** [review]

**Comment by sstok** [comment]
No tests? 🙀

**Comment by nicolas-grekas** [comment]
@sstok now with tests :)

**Review by sstok — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 1]
```diff
@@ -592,7 +597,15 @@ protected function initializeContainer()
         $this->container = require $cache->getPath();
         $this->container->set('kernel', $this);
 
-        if (!$fresh && $this->container->has('cache_warmer')) {
+        if ($fresh) {
+            return;
+        }
+
+        if ($oldContainer && get_class($this->container) !== $oldContainer->name) {
+            (new Filesystem())->remove(dirname($oldContainer->getFileName()));
```
Comment (stof) [inline src/Symfony/Component/HttpKernel/Kernel.php]: wouldn't this remove the whole cache dir ? This is not good, as it is not all about the DIC cache.

File: src/Symfony/Component/HttpKernel/Kernel.php [hunk 2]
```diff
@@ -584,6 +584,11 @@ protected function initializeContainer()
                     file_put_contents($this->getCacheDir().'/'.$class.'Compiler.log', null !== $container ? implode("\n", $container->getCompiler()->getLog()) : '');
                 }
             }
+
+            if ($oldContainer = file_exists($cache->getPath()) ? @include $cache->getPath() : false) {
+                $oldContainer = new \ReflectionClass($oldContainer);
```
Comment (umpirsky) [inline src/Symfony/Component/HttpKernel/Kernel.php]: @nicolas-grekas For me `$oldContainer` is `1` and getting:

```
[ReflectionException]   
  Class 1 does not exist
```

**Pull request merged**